x-pack/plugins/synthetics/server/routes/status/current_status.ts

```diff
@@ -0,0 +1,110 @@
+import { SavedObjectsClientContract } from '@kbn/core/server';
+import { SYNTHETICS_API_URLS } from '../../../common/constants';
+import { UMServerLibs } from '../../legacy_uptime/uptime_server';
+import { SyntheticsRestApiRouteFactory } from '../../legacy_uptime/routes';
+import { getMonitors } from '../util';
+import { getSnapshotCount } from '../../legacy_uptime/lib/requests/get_snapshot_counts';
+import { UptimeESClient } from '../../legacy_uptime/lib/lib';
+import { SyntheticsMonitorClient } from '../../synthetics_service/synthetics_monitor/synthetics_monitor_client';
+
+const DEFAULT_DATE_RANGE_START = 'now-1h';
+
+/**
+ * Helper function that converts a monitor's schedule to a value to use to generate
+ * an appropriate look-back window for snapshot count.
+ * @param schedule a number/unit pair that represents how often a configured monitor runs
+ * @returns schedule interval in ms
+ */
+export function periodToMs(schedule: { number: string; unit: Unit }) {
+  if (Object.keys(datemath.unitsMap).indexOf(schedule.unit) === -1) return 0;
+
+  return parseInt(schedule.number, 10) * datemath.unitsMap[schedule.unit].base;
+}
+
+function snapshotFilters(ids: Array<string | undefined>) {
+  return `{"terms": { "monitor.id": ${JSON.stringify(ids)}}}`;
+}
+
+export async function getCounts(
+  uptimeEsClient: UptimeESClient,
+  savedObjectsClient: SavedObjectsClientContract,
+  syntheticsMonitorClient: SyntheticsMonitorClient
+) {
+  let monitors;
+  const enabledIds: Array<string | undefined> = [];
+  const disabledIds: Array<string | undefined> = [];
+  let disabledCount = 0;
+  let page = 1;
+  let maxPeriod = 0;
+  /**
+   * Walk through all monitor saved objects, bucket IDs by disabled/enabled status.
+   *
+   * Track max period to make sure the snapshot query should reach back far enough to catch
+   * latest ping for all enabled monitors.
+   */
+  do {
+    monitors = await getMonitors(
+      {
+        perPage: 500,
+        page,
+        sortField: 'name.keyword',
+        sortOrder: 'asc',
+      },
+      syntheticsMonitorClient.syntheticsService,
+      savedObjectsClient
+    );
+    page++;
+    monitors.saved_objects.forEach((monitor) => {
+      if (monitor.attributes.enabled === false) {
+        disabledIds.push(monitor.id);
+        disabledCount += monitor.attributes.locations.length;
+      } else {
+        enabledIds.push(monitor.id);
+        maxPeriod = Math.max(maxPeriod, periodToMs(monitor.attributes.schedule));
+      }
+    });
+  } while (monitors.saved_objects.length === monitors.per_page);
+
+  // get a status count for all enabled monitors
+  const snapshot = await getSnapshotCount({
+    uptimeEsClient,
+    dateRangeEnd: 'now',
+    dateRangeStart: String(
+      datemath.parse('now')?.valueOf()! - maxPeriod ?? DEFAULT_DATE_RANGE_START
+    ),
+    filters: snapshotFilters(enabledIds),
+  });
+  return {
+    snapshot,
+    disabledCount,
+    disabledIds,
+    enabledIds,
```

Comment:
Is this used on the UI at all? I'm not sure if I would include it *necessarily*, only because for the snapshot query, it needs to be a combination of the `monitor.id` or `monitor.attributes[ConfigKey.CUSTOM_HEARTBEAT_ID]` to account for project monitors, and that might be a confusing if someone tried to reference this on the front end. I suppose it is important to return this for testing, however. Maybe just keep it off the redux state. Thoughts?

Reply:
Or change to something more semantic for additional clarity `enabledHeartbeatIds`.

Reply:
Thanks for pointing this out. I have created several fields that are coming down from the endpoint that likely won't be in the final commit we merge, it helps me with testing stuff out during development. I originally had intended for this PR to be reviewable a few days ago but hit a snag. I downgraded the status to "in progress" on the project board, but I didn't indicate that it was WIP on the PR. I spent a good chunk of time today revamping data fetching, and there are numerous testing steps I still need to add.